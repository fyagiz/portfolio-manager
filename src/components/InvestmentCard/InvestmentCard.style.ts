import { StyleSheet } from "react-native";
import { COLOR, STYLE } from "../../utils/constants";

export default StyleSheet.create({
  container: {
    margin: STYLE.margin,
    padding: STYLE.padding,
    borderRadius: STYLE.borderRadius,
  },
  innerContainer: {
    flexDirection: "row",
    justifyContent: "space-between",
  },
  textContainer: {
    flexDirection: "row",
  },
  profitText: {
    color: COLOR.profitColor,
  },
  lossText: {
    color: COLOR.lossColor,
  },
});
