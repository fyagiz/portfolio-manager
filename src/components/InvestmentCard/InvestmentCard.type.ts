import { GestureResponderEvent } from "react-native";

export type InvestmentCardPropsType = {
  investmentName: string;
  profit: string;
  profitPercentage: string;
  onPress?: (event: GestureResponderEvent) => void;
  testOnly_pressed?: boolean;
};
