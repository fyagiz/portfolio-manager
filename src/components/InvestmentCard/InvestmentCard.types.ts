import { GestureResponderEvent } from "react-native";

export type InvestmentCardProps = {
  investmentName: string;
  profit: string;
  profitPercentage: string;
  onPress?: (event: GestureResponderEvent) => void;
  testOnly_pressed?: boolean;
};
