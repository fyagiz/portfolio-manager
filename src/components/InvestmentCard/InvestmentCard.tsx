import { Pressable, Text } from "react-native";
import { InvestmentCardPropsType } from "./InvestmentCard.type";
import styles from "./InvestmentCard.style";

const InvestmentCard = (props: InvestmentCardPropsType) => {
  const { investmentName, profit, profitPercentage, onPress, testOnly_pressed } = props;
  return (
    <Pressable
      style={({ pressed }) => [
        styles.container,
        {
          backgroundColor: pressed ? "#F9DDA4" : "#F3CC79",
        },
      ]}
      onPress={onPress}
      testID="pressable"
      testOnly_pressed={testOnly_pressed}
    >
      <Text testID="investmentName">{investmentName}</Text>
      <Text testID="profit">{profit}</Text>
      <Text testID="profitPercentage">% {profitPercentage}</Text>
    </Pressable>
  );
};

export default InvestmentCard;
