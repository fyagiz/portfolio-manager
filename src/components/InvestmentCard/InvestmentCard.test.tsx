import { render, fireEvent } from "@testing-library/react-native";
import InvestmentCard from "./InvestmentCard";

/*
1- InvestmentCard should be rendered.
2- InvestmentCard should show investmentName.
3- InvestmentCard should show profit.
4- InvestmentCard should show profit percentage.
5- InvestmentCard should be pressed.
*/

describe("InvestmentCard Component Unit Tests", () => {
  test("should render correctly", () => {
    const investmentCardComponent = render(<InvestmentCard testID="test" investmentName="Test Fund" profit="5000 TL" profitPercentage="5" />);
    expect(investmentCardComponent).toMatchSnapshot();
  });

  test("should show investmentName", () => {
    const investmentCardComponent = render(<InvestmentCard testID="test" investmentName="Test Fund" profit="5000 TL" profitPercentage="5" />);
    const investmentCardChildren = investmentCardComponent.getByTestId("testInvestmentCardInvestmentName").children;
    const investmantName = investmentCardChildren.join("");
    expect(investmantName).toMatch("Test Fund");
  });

  test("should show profit", () => {
    const investmentCardComponent = render(<InvestmentCard testID="test" investmentName="Test Fund" profit="5000 TL" profitPercentage="5" />);
    const investmentCardChildren = investmentCardComponent.getByTestId("testInvestmentCardProfitNumber").children;
    const profit = investmentCardChildren.join("");
    expect(profit).toMatch("5000 TL");
  });

  test("should show profit percentage", () => {
    const investmentCardComponent = render(<InvestmentCard testID="test" investmentName="Test Fund" profit="5000 TL" profitPercentage="5" />);
    const investmentCardChildren = investmentCardComponent.getByTestId("testInvestmentCardPercentageNumber").children;
    const profitPercentage = investmentCardChildren.join("");
    expect(profitPercentage).toMatch("5");
  });

  test("should be pressed", () => {
    const mockFunction = jest.fn();
    const investmentCardComponent = render(
      <InvestmentCard testID="test" investmentName="Test Fund" profit="5000 TL" profitPercentage="5" onPress={mockFunction} testOnly_pressed />,
    );
    const pressable = investmentCardComponent.getByTestId("testInvestmentCardPressable");
    fireEvent(pressable, "click");
    expect(mockFunction).toBeCalled();
  });
});
