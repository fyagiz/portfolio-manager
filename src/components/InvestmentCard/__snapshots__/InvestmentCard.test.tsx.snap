// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InvestmentCard Component Unit Tests should render correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "borderRadius": 4,
        "margin": 4,
        "padding": 4,
      },
      {
        "backgroundColor": "#F3CC79",
      },
    ]
  }
  testID="pressable"
>
  <Text
    testID="investmentName"
  >
    Test Fund
  </Text>
  <Text
    testID="profit"
  >
    5000 TL
  </Text>
  <Text
    testID="profitPercentage"
  >
    % 
    5
  </Text>
</View>
`;
